modules/library/coverage/src/main/java/org/geotools/image/ImageWorker.java

```diff
@@ -4465,6 +4465,16 @@ private ROI mosaicROIs(Vector sources, ROI... roiArray) {
                 rois.add(r);
             }
         }
+        // handle possible null inputs
+        for (int i = 0; i < roiArray.length; i++) {
+            ROI roi = roiArray[i];
+            if(roi == null) {
+                RenderedImage img = (RenderedImage) sources.get(i);
+                ROI r = new ROIShape(new Rectangle(img.getMinX(), img.getMinY(), img.getWidth(), img.getHeight()));
```

Comment:
Wondering if would be better to use ROIGeometry instead.
However, in the previous lines we are still using ROIShape indeed, so no objection on using ROIShape in the new code too, to avoid mixing types.